munit/shared/src/main/scala/munit/Assertions.scala

```diff
@@ -288,6 +288,21 @@ trait Assertions extends MacroCompat.CompileErrorMacro {
     )
   }
 
+  private def exceptionHandlerFromAssertions(
+      assertions: Assertions,
+      clues: => Clues
+  ): ComparisonFailExceptionHandler =
+    new ComparisonFailExceptionHandler {
+      def handle(
+          message: String,
+          obtained: String,
+          expected: String,
+          loc: Location
+      ): Nothing = {
+        assertions.failComparison(message, obtained, expected, clues)(loc)
+      }
+    }
```

Comment:
I decided to extract this from `ComparisonFailExceptionHandler` so that the `scala-diff` package doesn't need to depend on `Assertions`